src/Protocol/ActivityPub.php

```diff
@@ -0,0 +1,1810 @@
+<?php
+/**
+ * @file src/Protocol/ActivityPub.php
+ */
+namespace Friendica\Protocol;
+
+use Friendica\Database\DBA;
+use Friendica\Core\System;
+use Friendica\BaseObject;
+use Friendica\Util\Network;
+use Friendica\Util\HTTPSignature;
+use Friendica\Core\Protocol;
+use Friendica\Model\Conversation;
+use Friendica\Model\Contact;
+use Friendica\Model\APContact;
+use Friendica\Model\Item;
+use Friendica\Model\Profile;
+use Friendica\Model\Term;
+use Friendica\Model\User;
+use Friendica\Util\DateTimeFormat;
+use Friendica\Util\Crypto;
+use Friendica\Content\Text\BBCode;
+use Friendica\Content\Text\HTML;
+use Friendica\Util\JsonLD;
+use Friendica\Util\LDSignature;
+use Friendica\Core\Config;
+
+/**
+ * @brief ActivityPub Protocol class
+ * The ActivityPub Protocol is a message exchange protocol defined by the W3C.
+ * https://www.w3.org/TR/activitypub/
+ * https://www.w3.org/TR/activitystreams-core/
+ * https://www.w3.org/TR/activitystreams-vocabulary/
+ *
+ * https://blog.joinmastodon.org/2018/06/how-to-implement-a-basic-activitypub-server/
+ * https://blog.joinmastodon.org/2018/07/how-to-make-friends-and-verify-requests/
+ *
+ * Digest: https://tools.ietf.org/html/rfc5843
+ * https://tools.ietf.org/html/draft-cavage-http-signatures-10#ref-15
+ *
+ * Mastodon implementation of supported activities:
+ * https://github.com/tootsuite/mastodon/blob/master/app/lib/activitypub/activity.rb#L26
+ *
+ * To-do:
+ *
+ * Receiver:
+ * - Update Note
+ * - Delete Note
+ * - Delete Person
+ * - Undo Announce
+ * - Reject Follow
+ * - Undo Accept
+ * - Undo Follow
+ * - Add
+ * - Create Image
+ * - Create Video
+ * - Event
+ * - Remove
+ * - Block
+ * - Flag
+ *
+ * Transmitter:
+ * - Announce
+ * - Undo Announce
+ * - Update Person
+ * - Reject Follow
+ * - Event
+ *
+ * General:
+ * - Attachments
+ * - nsfw (sensitive)
+ * - Queueing unsucessful deliveries
+ * - Polling the outboxes for missing content?
+ * - Possibly using the LD-JSON parser
+ */
+class ActivityPub
+{
+	const PUBLIC = 'https://www.w3.org/ns/activitystreams#Public';
```

Comment:
`public` is a reserved word in this context in PHP 5.6. Please rename it or bump the project's PHP version requirement to 7.

See https://travis-ci.org/friendica/friendica/jobs/434441655#L633

Reply:
Ah, okay.